### PR TITLE
copr: add coprocessor v2

### PR DESCRIPTION
**Pull request by Jibbow**
### What problem does this PR solve?

Issue Number: #9747

Problem Summary:

Setup a module for the new coprocessor framework.
Mostly I want to get some feedback whether I missed something obvious and about the naming of modules/structs.

### What is changed and how it works?

Proposal: [RFC: coprocessor plugin](https://github.com/andylokandy/rfcs/blob/plugin/text/2021-02-24-coprocessor-plugin.md) 

What's Changed:

### Related changes

- PR to update `kvproto`: https://github.com/pingcap/kvproto/pull/728

### Check List 

Tests 

- No code

### Release note

 * No release note

### CODE REVIEW

**Comment by Jibbow** [comment]
Requires https://github.com/pingcap/kvproto/pull/728 to be merged first.

Also: make sure to remove `[patch]` in `Cargo.toml`

**Review by skyzh — APPROVED** [review]
Looks good to me overall. Should merge after kvproto PR has been merged, as mentioned in previous comments.

**Comment by skyzh** [comment]
(Also a little reminder: you may use this command in PR comment `/cc @andylokandy @skyzh` to assign us as reviewers next time.)

File: src/server/server.rs
Comment (andylokandy) [inline src/server/server.rs:79]: Would it be better if it is `copr_v2: coprocessor_v2::Endpoint<E>`?

**Comment by andylokandy** [comment]
Generally looks to me, good job! For the `kvproto`, I'm calling for more reviews from other committers. Anyway, there is no problem  AFAICS.

**Review by zhongzc — COMMENTED** [review]
lgtm

File: src/server/server.rs
Comment (Jibbow) [reply src/server/server.rs]: Yes, good idea! Then the `Endpoint` struct name matches the filename. Changed it accordingly to `coprv2::Endpoint`, because the module is still called `coprv2`.

Comment (andylokandy) [reply src/server/server.rs]: Ahh, I see. Then why doesn't it named `coprocessor_v2` aligning with `coprocessor`

Comment (Jibbow) [reply src/server/server.rs]: Sure thing! I thought we said that we name the new module `coprv2`, but I like `coprocessor_v2` more anyway :)

**Comment by andylokandy** [comment]
@Jibbow Since the kvproto has been merged, we can then merge this one after the conflict is solved.

**Pull request merged**
